tests/test_i18n.py

```diff
@@ -44,6 +44,12 @@ def test_intcomma() -> None:
     try:
         humanize.i18n.activate("de_DE")
         assert humanize.intcomma(number) == "10.000.000"
+        assert humanize.intcomma(1_234_567.8901) == "1.234.567,8901"
+        assert humanize.intcomma(1_234_567.89) == "1.234.567,89"
+        assert humanize.intcomma("1234567,89") == "1.234.567,89"
+        assert humanize.intcomma("1.234.567,89") == "1.234.567,89"
+        assert humanize.intcomma("1.234.567,8") == "1.234.567,8"
+
         humanize.i18n.activate("fr_FR")
         assert humanize.intcomma(number) == "10 000 000"
```

Comment:
```
humanize.i18n.activate("EN")
assert humanize.intcomma(1_234_567.8901) == "1,234,567.8901"
```
It would be good to add a test for a non-German locale to show that the original behavior is preserved. @hugovk, are the doctests verified by the CI?

Also, it should be documented that if the input is a string, it must already match the locale. i.e. This function is not to be used to convert a string from one locale to another.

Reply:
> It would be good to add a test for a non-German locale

There are already [tests for the default locale](https://github.com/python-humanize/humanize/blob/d8e27393dbf4ed3645ffc3464c9c44f4d8e47534/tests/test_number.py#L34-L71). Or do you mean a non-German, non-default locale?

> Also, it should be documented that if the input is a string, it must already match the locale.

This was already the case with the thousands separator btw. Where would be a good place to document this?

Reply:
I'm not sure how best to write this, given that it doesn't really even mention that strings are allowed, except in two examples and partially in the arguments.

Reply:
> It would be good to add a test for a non-German locale to show that the original behavior is preserved. @hugovk, are the doctests verified by the CI?

Yes, doctests are run by pytest and pytest is run on the CI.